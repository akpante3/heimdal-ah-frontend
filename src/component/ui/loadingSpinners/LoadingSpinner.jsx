import React from 'react';
import { FadeLoader } from 'react-spinners';
import './LoadingSpinner.scss';

const override = `
  display: block;
  margin: 0 auto;
`;

const LoadingSpinner = ({ isLoading, size, color }) => (
  <div className="sweet-loading">
    <FadeLoader
      css={override}
      sizeUnit="px"
      size={size || 100}
      color={color || '#2C4D45'}
      loading={isLoading}
    />
  </div>
);

export default LoadingSpinner;
