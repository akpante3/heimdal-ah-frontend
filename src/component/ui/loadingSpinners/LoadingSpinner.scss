.sweet-loading {
  margin-left: 0px !important;
  margin-top: 0px !important;
}
