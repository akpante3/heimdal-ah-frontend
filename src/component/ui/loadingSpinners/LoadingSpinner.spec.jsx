/* eslint-disable no-undef */
import React from 'react';
import LoadingSpinner from './LoadingSpinner';

describe('The LoadingSpinner Component Test Suite', () => {
  it('should render the LoadingSpinner component', () => {
    const wrapper = shallow(<LoadingSpinner />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <LoadingSpinner /> Component', () => {
    const wrapper = shallow(<LoadingSpinner />);
    expect(wrapper.length).toBe(1);
    expect(wrapper.find('div')).toBeDefined();
    expect(wrapper.find('i')).toBeDefined();
  });
});
