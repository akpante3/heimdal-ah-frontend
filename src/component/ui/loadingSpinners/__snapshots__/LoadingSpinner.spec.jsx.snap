// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The LoadingSpinner Component Test Suite should render the LoadingSpinner component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadingSpinner />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <onlyUpdateForKeys(Loader)
        color="#2C4D45"
        css={
          Object {
            "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkxvYWRpbmdTcGlubmVyLmpzeCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFJb0IiLCJmaWxlIjoiTG9hZGluZ1NwaW5uZXIuanN4Iiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCB7IGNzcyB9IGZyb20gJ0BlbW90aW9uL2NvcmUnO1xuaW1wb3J0IHsgRG90TG9hZGVyIH0gZnJvbSAncmVhY3Qtc3Bpbm5lcnMnO1xuXG5jb25zdCBvdmVycmlkZSA9IGNzc2BcbiAgZGlzcGxheTogYmxvY2s7XG4gIG1hcmdpbjogMCBhdXRvO1xuICBib3JkZXItY29sb3I6IHJlZDtcbmA7XG5cbmNvbnN0IExvYWRpbmdTcGlubmVyID0gKHsgaXNMb2FkaW5nIH0pID0+IChcbiAgPGRpdiBjbGFzc05hbWU9XCJjb2wtbWQtMTIgdGV4dC1jZW50ZXIgc3dlZXQtbG9hZGluZ1wiPlxuICAgIDxEb3RMb2FkZXIgY3NzPXtvdmVycmlkZX0gc2l6ZVVuaXQ9XCJweFwiIHNpemU9ezE1MH0gY29sb3I9XCIjMkM0RDQ1XCIgbG9hZGluZz17aXNMb2FkaW5nfSAvPlxuICA8L2Rpdj5cbik7XG5cbmV4cG9ydCBkZWZhdWx0IExvYWRpbmdTcGlubmVyO1xuIl19 */",
            "name": "3vn6pu-override",
            "styles": "display:block;margin:0 auto;border-color:red;label:override;",
          }
        }
        loading={true}
        size={150}
        sizeUnit="px"
      />,
      "className": "col-md-12 text-center sweet-loading",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "color": "#2C4D45",
        "css": Object {
          "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkxvYWRpbmdTcGlubmVyLmpzeCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFJb0IiLCJmaWxlIjoiTG9hZGluZ1NwaW5uZXIuanN4Iiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCB7IGNzcyB9IGZyb20gJ0BlbW90aW9uL2NvcmUnO1xuaW1wb3J0IHsgRG90TG9hZGVyIH0gZnJvbSAncmVhY3Qtc3Bpbm5lcnMnO1xuXG5jb25zdCBvdmVycmlkZSA9IGNzc2BcbiAgZGlzcGxheTogYmxvY2s7XG4gIG1hcmdpbjogMCBhdXRvO1xuICBib3JkZXItY29sb3I6IHJlZDtcbmA7XG5cbmNvbnN0IExvYWRpbmdTcGlubmVyID0gKHsgaXNMb2FkaW5nIH0pID0+IChcbiAgPGRpdiBjbGFzc05hbWU9XCJjb2wtbWQtMTIgdGV4dC1jZW50ZXIgc3dlZXQtbG9hZGluZ1wiPlxuICAgIDxEb3RMb2FkZXIgY3NzPXtvdmVycmlkZX0gc2l6ZVVuaXQ9XCJweFwiIHNpemU9ezE1MH0gY29sb3I9XCIjMkM0RDQ1XCIgbG9hZGluZz17aXNMb2FkaW5nfSAvPlxuICA8L2Rpdj5cbik7XG5cbmV4cG9ydCBkZWZhdWx0IExvYWRpbmdTcGlubmVyO1xuIl19 */",
          "name": "3vn6pu-override",
          "styles": "display:block;margin:0 auto;border-color:red;label:override;",
        },
        "loading": true,
        "size": 150,
        "sizeUnit": "px",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <onlyUpdateForKeys(Loader)
          color="#2C4D45"
          css={
            Object {
              "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkxvYWRpbmdTcGlubmVyLmpzeCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFJb0IiLCJmaWxlIjoiTG9hZGluZ1NwaW5uZXIuanN4Iiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCB7IGNzcyB9IGZyb20gJ0BlbW90aW9uL2NvcmUnO1xuaW1wb3J0IHsgRG90TG9hZGVyIH0gZnJvbSAncmVhY3Qtc3Bpbm5lcnMnO1xuXG5jb25zdCBvdmVycmlkZSA9IGNzc2BcbiAgZGlzcGxheTogYmxvY2s7XG4gIG1hcmdpbjogMCBhdXRvO1xuICBib3JkZXItY29sb3I6IHJlZDtcbmA7XG5cbmNvbnN0IExvYWRpbmdTcGlubmVyID0gKHsgaXNMb2FkaW5nIH0pID0+IChcbiAgPGRpdiBjbGFzc05hbWU9XCJjb2wtbWQtMTIgdGV4dC1jZW50ZXIgc3dlZXQtbG9hZGluZ1wiPlxuICAgIDxEb3RMb2FkZXIgY3NzPXtvdmVycmlkZX0gc2l6ZVVuaXQ9XCJweFwiIHNpemU9ezE1MH0gY29sb3I9XCIjMkM0RDQ1XCIgbG9hZGluZz17aXNMb2FkaW5nfSAvPlxuICA8L2Rpdj5cbik7XG5cbmV4cG9ydCBkZWZhdWx0IExvYWRpbmdTcGlubmVyO1xuIl19 */",
              "name": "3vn6pu-override",
              "styles": "display:block;margin:0 auto;border-color:red;label:override;",
            }
          }
          loading={true}
          size={150}
          sizeUnit="px"
        />,
        "className": "col-md-12 text-center sweet-loading",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "#2C4D45",
          "css": Object {
            "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkxvYWRpbmdTcGlubmVyLmpzeCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFJb0IiLCJmaWxlIjoiTG9hZGluZ1NwaW5uZXIuanN4Iiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCB7IGNzcyB9IGZyb20gJ0BlbW90aW9uL2NvcmUnO1xuaW1wb3J0IHsgRG90TG9hZGVyIH0gZnJvbSAncmVhY3Qtc3Bpbm5lcnMnO1xuXG5jb25zdCBvdmVycmlkZSA9IGNzc2BcbiAgZGlzcGxheTogYmxvY2s7XG4gIG1hcmdpbjogMCBhdXRvO1xuICBib3JkZXItY29sb3I6IHJlZDtcbmA7XG5cbmNvbnN0IExvYWRpbmdTcGlubmVyID0gKHsgaXNMb2FkaW5nIH0pID0+IChcbiAgPGRpdiBjbGFzc05hbWU9XCJjb2wtbWQtMTIgdGV4dC1jZW50ZXIgc3dlZXQtbG9hZGluZ1wiPlxuICAgIDxEb3RMb2FkZXIgY3NzPXtvdmVycmlkZX0gc2l6ZVVuaXQ9XCJweFwiIHNpemU9ezE1MH0gY29sb3I9XCIjMkM0RDQ1XCIgbG9hZGluZz17aXNMb2FkaW5nfSAvPlxuICA8L2Rpdj5cbik7XG5cbmV4cG9ydCBkZWZhdWx0IExvYWRpbmdTcGlubmVyO1xuIl19 */",
            "name": "3vn6pu-override",
            "styles": "display:block;margin:0 auto;border-color:red;label:override;",
          },
          "loading": true,
          "size": 150,
          "sizeUnit": "px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
